master/internal/trial.go

```diff
@@ -364,57 +373,110 @@ func (t *trial) runningReceive(ctx *actor.Context) error {
 		}
 
 	case actor.ChildFailed:
-		ctx.Tell(t.rp, scheduler.TerminateTask{TaskID: t.task.ID, Forcible: true})
+		t.terminate(ctx, true)
 
+	case killTrial:
+		t.killed = true
+		t.terminate(ctx, true)
+
+	case allReadyTimeout:
+		if msg.runID == t.runID &&
+			time.Now().After(t.lastContainerConnectedTime.Add(allReadyTimeoutPeriod)) {
+			ctx.Tell(
+				t.logger,
+				model.TrialLog{
+					TrialID: t.id,
+					Message: "some containers are taking a long time to connect to master; " +
```

Comment:
non-blocking: this indentation seems a bit off but could be just github rendering